### PR TITLE
Bug 1823306 - add null check for bottomToolbarChangedAction

### PR DESCRIPTION
**Pull request by HarrisonOg**
Prospective fix for [cannot round NaN in `EngineViewBrowserToolbarBehavior`](https://bugzilla.mozilla.org/show_bug.cgi?id=1823306).

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1823306

### CODE REVIEW

**Comment by HarrisonOg** [comment]
I could not reproduce the crash, this a prospective fix for this issue.

`Float.NaN` is a constant that can occasionally be passed to `Float` values but will cause issue with rounding functions.

File: android-components/components/feature/session/src/main/java/mozilla/components/feature/session/behavior/EngineViewBrowserToolbarBehavior.kt [hunk 1]
```diff
@@ -41,7 +41,9 @@ class EngineViewBrowserToolbarBehavior(
     @VisibleForTesting
     internal var toolbarChangedAction: (Float) -> Unit?
     private val bottomToolbarChangedAction = { newToolbarTranslationY: Float ->
-        engineView?.setVerticalClipping(-newToolbarTranslationY.roundToInt())
+        if (!newToolbarTranslationY.isNaN()) {
```
Comment (t-p-white) [inline android-components/components/feature/session/src/main/java/mozilla/components/feature/session/behavior/EngineViewBrowserToolbarBehavior.kt:44]: is it possible to also add a test for this condition?

**Review by t-p-white — COMMENTED** [review]
lgtm, just a question around tests

File: android-components/components/feature/session/src/main/java/mozilla/components/feature/session/behavior/EngineViewBrowserToolbarBehavior.kt (hunk 1 above)
Comment (HarrisonOg) [reply android-components/components/feature/session/src/main/java/mozilla/components/feature/session/behavior/EngineViewBrowserToolbarBehavior.kt:44]: I'll check into it

**Comment by HarrisonOg** [comment]
Added test for NaN condition

**Review by t-p-white — APPROVED** [review]
Thanks for adding the test :shipit:

**Comment by HarrisonOg** [comment]
@mergify backport releases_v119

**Pull request merged**